src/front/wgsl/lexer.rs

```diff
@@ -356,9 +356,41 @@ fn consume_token(mut input: &str, generic: bool) -> (Token<'_>, &str) {
                 (Token::UnterminatedString, quote_content)
             }
         }
-        '/' if chars.as_str().starts_with('/') => {
-            let _ = chars.position(|c| c == '\n' || c == '\r');
-            (Token::Trivia, chars.as_str())
+        '/' => {
+            input = chars.as_str();
+            match chars.next() {
+                Some('/') => {
+                    let _ = chars.position(|c| c == '\n' || c == '\r');
+                    (Token::Trivia, chars.as_str())
+                }
+                Some('*') => {
+                    input = chars.as_str();
+
+                    let mut depth = 1;
+                    while let Some(c) = chars.next() {
+                        match c {
+                            '*' if chars.next() == Some('/') => {
```

Comment:
this will fail on a case like this: `/* blabla **/`

Reply:
hopefully it's fixed, there can be strange cases like `/*/**/****/`, which is a valid closed comment